fmt/printf.h

```diff
@@ -92,9 +110,18 @@ class ArgConverter : public ArgVisitor<ArgConverter<T>, void> {
       visit_any_int(value);
   }
 
+  void visit_char(char value) {
+    if (type_ != 's')
+      visit_any_int(value);
+  }
```

Comment:
ArgConverter runs before DefaultType, and I couldn't see a clean way to switch it around.